state/migration_export.go

```diff
@@ -2220,7 +2220,8 @@ func (e *exporter) getCharmOrigin(doc applicationDoc, defaultArch string) (descr
 	origin := doc.CharmOrigin
 
 	// If the channel is empty, then we fall back to the Revision.
-	var revision int
+	// Default to -1 since that is used to commonly denote no revision.
+	revision := -1
```

Comment:
-1 should only be seen in the client before a charm is deployed. This should be fixed on the import side, not export. During the import, you can find the revision for the charm and add it. We have code to create charm origins in the import, or should due to <2.9 migrations